### PR TITLE
emscripten: 3.0.0 -> 3.1.12

### PR DESCRIPTION
**Pull request by yorickvP**
###### Description of changes

Update emscripten and its dependencies, also fix build dependencies (were complaining about missing archive indices).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yorickvP** [comment]
@ofborg build emscripten emscriptenPackages.xmlmirror

File: pkgs/development/compilers/binaryen/default.nix [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
-  version = "102";
+  version = "108";
 
   src = fetchFromGitHub {
     owner = "WebAssembly";
     repo = "binaryen";
     rev = "version_${version}";
-    sha256 = "sha256-UlktpY9tyjYNkmiBZM42QGg67kcPo7VDy2B4Ty1YIew=";
+    sha256 = "sha256-CcGxPBdUiNLfMjjJKFMdDvaIrHvf2M/gCub4JBw4+8c=";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/binaryen/default.nix:12]: We should use our googletest instead.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 175358` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>binaryen</li>
    <li>emscripten</li>
    <li>emscriptenPackages.json_c</li>
    <li>emscriptenPackages.libxml2</li>
    <li>emscriptenPackages.xmlmirror</li>
    <li>emscriptenPackages.zlib</li>
  </ul>
</details>

File: pkgs/development/compilers/binaryen/default.nix (hunk 1 above)
Comment (yorickvP) [reply pkgs/development/compilers/binaryen/default.nix:12]: I don't think we have a googletest?

Comment (yorickvP) [reply pkgs/development/compilers/binaryen/default.nix:12]: Update: we have a gtest, but the package will require some patching to use this

**Comment by yorickvP** [comment]
@ofborg build binaryen

File: pkgs/development/compilers/binaryen/default.nix (hunk 1 above)
Comment (yorickvP) [reply pkgs/development/compilers/binaryen/default.nix]: Added this and enabled tests.

File: pkgs/development/compilers/binaryen/default.nix [hunk 2]
```diff
@@ -1,18 +1,42 @@
...
+    "crash" "dylink" "ctor-eval"
+    "wasm-metadce" "wasm-reduce" "spec"
+    "lld" "wasm2js" "validator"
+    "example" "unit"
+    # "binaryenjs" "binaryenjs_wasm" # not building this
+    "lit" "gtest"
+  ];
+  doCheck = true;
+
   meta = with lib; {
```
Comment (ckiee) [inline pkgs/development/compilers/binaryen/default.nix:40]: Ofborg failed on *-darwin, can you mark as broken or debug?

File: pkgs/development/compilers/binaryen/default.nix [hunk 3]
```diff
@@ -1,18 +1,42 @@
...
+  preConfigure = ''
+    if [ $doCheck -eq 1 ]; then
+      sed -i '/googletest/d' third_party/CMakeLists.txt
+    else
+      cmakeFlagsArray=($cmakeFlagsArray -DBUILD_TESTS=0)
+    fi
+  '';
+
+  checkInputs = [ gtest lit nodejs filecheck ];
+  checkPhase = "LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/lib python3 ../check.py $tests";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/binaryen/default.nix:27]:
```suggestion
  checkPhase = ''
    LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/lib python3 ../check.py $tests
  '';
```

File: pkgs/development/python-modules/filecheck/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  };
+
+  meta = with lib; {
+    homepage = "https://pypi.org/project/filecheck/";
+    license = licenses.asl20;
+    description = "Python port of LLVM's FileCheck, flexible pattern matching file verifier";
+    maintainers = with maintainers; [ yorickvp ];
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/filecheck/default.nix:22]:
```suggestion
  };
}
```

File: pkgs/development/python-modules/filecheck/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+buildPythonApplication rec {
+  pname = "filecheck";
+  version = "0.0.22";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-Ae4ye2t0LO0v2eRDGdLXBqRXPl4O5yYZONDeFqd+XAY=";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/filecheck/default.nix:14]: Missing pythonImportsCheck

File: pkgs/development/python-modules/filecheck/default.nix [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  meta = with lib; {
+    homepage = "https://pypi.org/project/filecheck/";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/filecheck/default.nix:16]:
```suggestion
    homepage = "https://github.com/mull-project/FileCheck.py";
```

**Comment by gracicot** [comment]
The checks for binaryen don't seem to work on darwin-x86_64. Here's the error output:
```
[ running validation tests... ]

executing:  /private/tmp/nix-build-binaryen-108.drv-0/source/build/bin/wasm-as /private/tmp/nix-build-binaryen-108.drv-0/source/test/validator/invalid_export.wast -o a.wasm
executing:  /private/tmp/nix-build-binaryen-108.drv-0/source/build/bin/wasm-as /private/tmp/nix-build-binaryen-108.drv-0/source/test/validator/invalid_import.wast -o a.wasm
executing:  /private/tmp/nix-build-binaryen-108.drv-0/source/build/bin/wasm-as --validate=web /private/tmp/nix-build-binaryen-108.drv-0/source/test/validator/invalid_export.wast -o a.wasm
[wasm-validator error in module] i64 == i64: Exported function must not have i64 results, on
export64
Fatal: Error: input module is not valid.

executing:  /private/tmp/nix-build-binaryen-108.drv-0/source/build/bin/wasm-as --validate=web /private/tmp/nix-build-binaryen-108.drv-0/source/test/validator/invalid_import.wast -o a.wasm
[wasm-validator error in module] i64 == i64: Imported function must not have i64 parameters, on
bad
Fatal: Error: input module is not valid.

executing:  /private/tmp/nix-build-binaryen-108.drv-0/source/build/bin/wasm-as --validate=none /private/tmp/nix-build-binaryen-108.drv-0/source/test/validator/invalid_return.wast -o a.wasm
executing:  /private/tmp/nix-build-binaryen-108.drv-0/source/build/bin/wasm-as /private/tmp/nix-build-binaryen-108.drv-0/source/test/validator/invalid_number.wast -o a.wasm
[parse exception: invalid i64 format]Fatal: error in parsing input

[ checking native example testcases...]

build:  clang /private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-hello-world.c -c -o example.o -I/private/tmp/nix-build-binaryen-108.drv-0/source/src -g -L/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib -pthread
clang-11: warning: argument unused during compilation: '-L/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib' [-Wunused-command-line-argument]
   /private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-hello-world.c /private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-hello-world.c /private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-hello-world.txt
link:  clang++ -std=c++17 example.o -L/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib -lbinaryen -I/private/tmp/nix-build-binaryen-108.drv-0/source/t -g -pthread -o example -Wl,-rpath,/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib
run... example
build:  clang /private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c -c -o example.o -I/private/tmp/nix-build-binaryen-108.drv-0/source/src -g -L/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib -pthread
clang-11: warning: argument unused during compilation: '-L/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib' [-Wunused-command-line-argument]
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:190:41: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeNone: %d\n", none);
                                 ~~     ^~~~
                                 %lu
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:196:48: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeUnreachable: %d\n", unreachable);
                                        ~~     ^~~~~~~~~~~
                                        %lu
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:202:42: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeInt32: %d\n", i32);
                                  ~~     ^~~
                                  %lu
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:208:42: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeInt64: %d\n", i64);
                                  ~~     ^~~
                                  %lu
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:214:44: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeFloat32: %d\n", f32);
                                    ~~     ^~~
                                    %lu
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:220:44: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeFloat64: %d\n", f64);
                                    ~~     ^~~
                                    %lu
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:226:43: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeVec128: %d\n", v128);
                                   ~~     ^~~~
                                   %lu
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:232:44: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeFuncref: %d\n", funcref);
                                    ~~     ^~~~~~~
                                    %lu
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:238:46: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeExternref: %d\n", externref);
                                      ~~     ^~~~~~~~~
                                      %lu
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:244:43: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeAnyref: %d\n", anyref);
                                   ~~     ^~~~~~
                                   %lu
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:250:42: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeEqref: %d\n", eqref);
                                  ~~     ^~~~~
                                  %lu
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:256:43: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeI31ref: %d\n", i31ref);
                                   ~~     ^~~~~~
                                   %lu
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:262:44: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeDataref: %d\n", dataref);
                                    ~~     ^~~~~~~
                                    %lu
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c:267:41: warning: format specifies type 'int' but the argument has type 'BinaryenType' (aka 'unsigned long') [-Wformat]
  printf("  // BinaryenTypeAuto: %d\n", BinaryenTypeAuto());
                                 ~~     ^~~~~~~~~~~~~~~~~~
                                 %lu
14 warnings generated.
   /private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c /private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.c /private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-kitchen-sink.txt
link:  clang++ -std=c++17 example.o -L/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib -lbinaryen -I/private/tmp/nix-build-binaryen-108.drv-0/source/t -g -pthread -o example -Wl,-rpath,/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib
run... example
[wasm-validator error in function func] local.set's value type must be correct, on
(local.set $0
 (i64.const 1234)
)
build:  clang /private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-multiple-tables.c -c -o example.o -I/private/tmp/nix-build-binaryen-108.drv-0/source/src -g -L/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib -pthread
clang-11: warning: argument unused during compilation: '-L/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib' [-Wunused-command-line-argument]
   /private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-multiple-tables.c /private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-multiple-tables.c /private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-multiple-tables.txt
link:  clang++ -std=c++17 example.o -L/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib -lbinaryen -I/private/tmp/nix-build-binaryen-108.drv-0/source/t -g -pthread -o example -Wl,-rpath,/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib
run... example
build:  clang /private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-relooper-unreachable-if.cpp -c -o example.o -I/private/tmp/nix-build-binaryen-108.drv-0/source/src -g -L/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib -pthread -std=c++17
clang-11: warning: argument unused during compilation: '-L/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib' [-Wunused-command-line-argument]
/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-relooper-unreachable-if.cpp:4:10: fatal error: 'cassert' file not found
#include <cassert>
         ^~~~~~~~~
1 error generated.
Traceback (most recent call last):
  File "/private/tmp/nix-build-binaryen-108.drv-0/source/build/../check.py", line 416, in <module>
    sys.exit(main())
  File "/private/tmp/nix-build-binaryen-108.drv-0/source/build/../check.py", line 399, in main
    TEST_SUITES[test]()
  File "/private/tmp/nix-build-binaryen-108.drv-0/source/build/../check.py", line 301, in run_example_tests
    subprocess.check_call(extra)
  File "/nix/store/lfayq1vz3szk4ji0s53v0x1ijcdn3qnx-python3-3.9.12/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['clang', '/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-relooper-unreachable-if.cpp', '-c', '-o', 'example.o', '-I/private/tmp/nix-build-binaryen-108.drv-0/source/src', '-g', '-L/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib', '-pthread', '-std=c++17']' returned non-zero exit status 1.
error: builder for '/nix/store/s7irmilv2768w6dd9j6d7d04plrqdy60-binaryen-108.drv' failed with exit code 1;
       last 10 log lines:
       > Traceback (most recent call last):
       >   File "/private/tmp/nix-build-binaryen-108.drv-0/source/build/../check.py", line 416, in <module>
       >     sys.exit(main())
       >   File "/private/tmp/nix-build-binaryen-108.drv-0/source/build/../check.py", line 399, in main
       >     TEST_SUITES[test]()
       >   File "/private/tmp/nix-build-binaryen-108.drv-0/source/build/../check.py", line 301, in run_example_tests
       >     subprocess.check_call(extra)
       >   File "/nix/store/lfayq1vz3szk4ji0s53v0x1ijcdn3qnx-python3-3.9.12/lib/python3.9/subprocess.py", line 373, in check_call
       >     raise CalledProcessError(retcode, cmd)
       > subprocess.CalledProcessError: Command '['clang', '/private/tmp/nix-build-binaryen-108.drv-0/source/test/example/c-api-relooper-unreachable-if.cpp', '-c', '-o', 'example.o', '-I/private/tmp/nix-build-binaryen-108.drv-0/source/src', '-g', '-L/private/tmp/nix-build-binaryen-108.drv-0/source/build/lib', '-pthread', '-std=c++17']' returned non-zero exit status 1.
       For full logs, run 'nix log /nix/store/s7irmilv2768w6dd9j6d7d04plrqdy60-binaryen-108.drv'.
error: 1 dependencies of derivation '/nix/store/j6zgrvfff66rw6a2grkii7xzpjg1ni60-emscripten-3.1.12.drv' failed to build
error: build of '/nix/store/6hqrmkmdnbhgni36d7w219pf1958vknh-npm-8.11.0.drv', '/nix/store/j6zgrvfff66rw6a2grkii7xzpjg1ni60-emscripten-3.1.12.drv' failed
```

**Comment by SuperSandro2000** [comment]
> The checks for binaryen don't seem to work on darwin-x86_64. Here's the error output:

If only checks are broken I would suggest to disable them on darwin. If the entire binary is broken then please mark it broken for darwin.

**Comment by SuperSandro2000** [comment]
If no one can help here marking it broken is fine @NixOS/darwin-maintainers

File: pkgs/development/python-modules/filecheck/default.nix (hunk 5 above)
Comment (yorickvP) [reply pkgs/development/python-modules/filecheck/default.nix:14]: The pypi source does not package the tests.

Comment (prusnak) [reply pkgs/development/python-modules/filecheck/default.nix:14]: If that's the case, please use fetchFromGitHub instead of fetchPypi.

Comment (yorickvP) [reply pkgs/development/python-modules/filecheck/default.nix:14]: I would strongly prefer not to, the github repo also uses poetry instead of setuptools.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/filecheck/default.nix:14]: Changing to github source is a standard procedure to get tests executed which we should always do if they work in the sandbox.

Building from poetry is easy. Add ``format = "pyproject";`` and check if the project uses poetry or poetry-core and add that to nativeBuildInputs.

**Comment by prusnak** [comment]
> re there different tiers support (like tier1 = linux amd64, tier 2 = linux arm64, tier 3 = darwin) that would allow to merge a commit breaking a lower tier at the benefit of an upper tier.

Yes, these are defined https://github.com/NixOS/rfcs/blob/master/rfcs/0046-platform-support-tiers.md

**Comment by toonn** [comment]
Fwiw, this looks like something for upstream to at least investigate. They may be making assumptions based on Linux-specific behavior here.

**Comment by rapenne-s** [comment]
> > re there different tiers support (like tier1 = linux amd64, tier 2 = linux arm64, tier 3 = darwin) that would allow to merge a commit breaking a lower tier at the benefit of an upper tier.
> 
> Yes, these are defined https://github.com/NixOS/rfcs/blob/master/rfcs/0046-platform-support-tiers.md

thanks! Given this document, we need to involve @NixOS/darwin-maintainers to know if the update can be merged or not.

**Comment by willcohen** [comment]
Per https://github.com/emscripten-core/emscripten/issues/10072, emscripten expects tip-of-tree llvm. I see that you bumped llvm to 14 -- is there any chance that this is a yet-unresolved llvm issue? It strikes me as generally unlikely, but at the moment I can't figure out why this error would be re-cropping up under llvm 14, and @gracicot's already followed up with all the obvious leads for similar issues.

**Comment by gracicot** [comment]
@willcohen The only way I can get clang to have a `std::max_align_t` of `8` is when it's compiling for a 32 bit target. It makes sense as I think the target emscripten is using for web assembly is a 32 bit target (wasm32-unknown-unknown). Maybe the target argument is not sent correctly to the compiler?

**Comment by willcohen** [comment]
If there's a pressing reason to get this merged, I suppose okay to break it on darwin for now (though I was finally getting the emscriptenPackages to build again on darwin as of a few weeks ago!). That said, if there's nothing pressing, it'd be great to get another week or two to work on this. I feel odd posing such an open-ended question upstream and suspect that we need to do some bisecting to figure out what broke this -- at least what point release and hopefully what commit -- but I will need probably at least another week to find the time to get into that in a little more detail.

**Review by willcohen — APPROVED** [review]

**Comment by gracicot** [comment]
@willcohen I confirm I'm able to build it. I'm not sure what went wrong in my previous attempt.

**Comment by SuperSandro2000** [comment]
> Would it work to merge #172741 instead? Then at least we'd be up to 3.1.10 and can work on addressing the LLVM issues needed to get to 3.1.12 after that in this PR after rebasing?

merged

**Comment by willcohen** [comment]
Thanks! Now that we've determined that this PR should be okay, looks like this'll just require a rebase to bump from 3.1.10 and include the other `emscriptenStdenv` fixes, etc!

**Comment by willcohen** [comment]
I've rebased this and updated it to 3.1.13 with #178620. Note that 3.1.14 breaks due to https://github.com/NixOS/nixpkgs/pull/178620#issuecomment-1163539397.

**Pull request closed without merging**